CMakeLists.txt

```diff
@@ -71,13 +66,47 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     endif()
 endif()
 
-if(NOT TARGET json-hpp) # if used as a subdirectory do not install json-schema.hpp
+if(ADD_INSTALL_TARGET) # if used as a subdirectory do not install json-schema.hpp
+    include(GenerateExportHeader)
+    generate_export_header(${PROJECT_NAME})
+    set_property(TARGET ${PROJECT_NAME} PROPERTY VERSION ${PROJECT_VERSION})
+    set_property(TARGET ${PROJECT_NAME} PROPERTY SOVERSION 1)
+    set_property(TARGET ${PROJECT_NAME} PROPERTY
+      INTERFACE_${PROJECT_NAME}_MAJOR_VERSION 1)
+    set_property(TARGET ${PROJECT_NAME} APPEND PROPERTY
+      COMPATIBLE_INTERFACE_STRING ${PROJECT_NAME}_MAJOR_VERSION)
+    
+    install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}Targets
+      LIBRARY DESTINATION lib
+      ARCHIVE DESTINATION lib
+      RUNTIME DESTINATION bin
+      INCLUDES DESTINATION include)
+    install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/json-schema.hpp
+        DESTINATION include
+        COMPONENT Devel)
+
+    include(CMakePackageConfigHelpers)
+    set(ConfigPackageLocation lib/cmake/${PROJECT_NAME})
+    write_basic_package_version_file(
+        "${CMAKE_CURRENT_BINARY_DIR}/cmake-config/${PROJECT_NAME}ConfigVersion.cmake"
+        VERSION ${PROJECT_VERSION}
+        COMPATIBILITY AnyNewerVersion)
+    export(EXPORT "${PROJECT_NAME}Targets"
+        FILE "${CMAKE_CURRENT_BINARY_DIR}/cmake-config/${PROJECT_NAME}Targets.cmake"
+        NAMESPACE ${PROJECT_NAME}::)
+    configure_package_config_file(cmake/config.cmake.in
+        "${CMAKE_CURRENT_BINARY_DIR}/cmake-config/${PROJECT_NAME}Config.cmake"
+        INSTALL_DESTINATION ${ConfigPackageLocation})
+    install(EXPORT "${PROJECT_NAME}Targets"
+        FILE ${PROJECT_NAME}Targets.cmake
+        NAMESPACE ${PROJECT_NAME}::
+        DESTINATION ${ConfigPackageLocation})
     install(
         FILES
-           ${CMAKE_CURRENT_SOURCE_DIR}/src/json-schema.hpp
-        DESTINATION
-            ${CMAKE_INSTALL_PREFIX}/include
-    )
+            "${CMAKE_CURRENT_BINARY_DIR}/cmake-config/${PROJECT_NAME}Config.cmake"
+            "${CMAKE_CURRENT_BINARY_DIR}/cmake-config/${PROJECT_NAME}ConfigVersion.cmake"
+        DESTINATION ${ConfigPackageLocation}
+        COMPONENT Devel)
 endif()
```

Comment:
Probably. I pulled this from cmake's documentation. Since this, I've implemented these imports a few more times, and I've found some of these commands seem a bit redundant or unnecessary, but from the documentation it's not clear to me what's actually required and what's not, and what's expected practice. For such a common function it's horribly documented (which is probably why every cmake package has it's own uniquely frustrating and opaque import semantics).